src/routes/[[lang=lang]]/documents/+page.server.ts

```diff
@@ -62,6 +82,25 @@ export const load: PageServerLoad = async ({ locals, url }) => {
       });
       break;
   }
+
+  const oldSRDGroupedByMeeting = oldFormatSRDFiles.reduce<
+    Record<string, FileData[]>
+  >((acc, file) => {
+    const fileParts = file.id.split("/");
+    const fileName = fileParts[fileParts.length - 1] ?? "unknown";
+    const meeting = fileName!.substring(
+      0,
+      fileName.endsWith("pdf")
+        ? fileName.length - 3
+        : fileName.endsWith("html")
+          ? fileName.length - 4
+          : fileName.length,
+    );
```

Comment:
I feel like nested ternarys could be confusing here. Also, if we were to handle more file formats it would not be easily expandable for that logic.

Also, is fileName not always defined? If so I don't think we need the non-null assertion ("!").

Maybe something like this? (I haven't tested it)
```
const extensions = ["pdf", "html"]
const meeting = extensions.some((ext) => {
  if (fileName.endsWith(ext)) return fileName.substring(0, fileName.length - ext.length)
}) ?? fileName
```

Reply:
Good point, just had to modify your code a bit